### PR TITLE
Dt snapshot command

### PR DESCRIPTION
**Pull request by dtafesse**

### CODE REVIEW

File: project/eScavenge/StringConst.py [hunk 1]
```diff
@@ -23,10 +23,10 @@
 edit_landmark_fail = "Landmark Not Edited Successfully"
 edit_landmark_order_success = "Landmark Order Edited Successfully"
 edit_landmark_order_fail = "Landmark Order Not Edited Successfully"
+no_team_logged_in = "There is no Team Logged In"
```
Comment (Purlieu) [inline project/eScavenge/StringConst.py]: Only thing I can see that I can comment on is just that this instance of no_team_logged_in isn't used anywhere, but I think that came from an old commit of mine

File: project/eScavenge/CLI.py [hunk 2]
```diff
@@ -260,6 +260,15 @@ def get_stats(self, args):
     return "You cannot see another users stats"
 
 
+@need_admin
+def get_snapshot(self, _):
+    if not self.game.started or self.game.ended:
```
Comment (holyzala) [inline project/eScavenge/CLI.py]: the game state check should be done in game

File: project/eScavenge/Game.py [hunk 3]
```diff
@@ -268,7 +268,17 @@ def get_status(self, now, username):
...
+        for username in self.__teams:
+            current_team = self.get_team(username)
+            total_time = datetime.timedelta(days=0, hours=0, minutes=0, seconds=0)
+            for t in current_team.time_log:
+                total_time += t
+            if current_team.current_landmark <= len(self.__landmarks):
+                stat_str = "Team: {}\nYou Are On Landmark {}\nTime Taken For Landmarks: {}\nTotal Points: {}\n"
+                stringList.append(stat_str.format(username, current_team.current_landmark + 1, total_time,
+                                                  current_team.points))
+        return ''.join(stringList)
```
Comment (holyzala) [inline project/eScavenge/Game.py]: This is just returning a string, which is fine, but to address my comment about checking state in game you can return both an error and a string. so this line would be `return Errors.SUCCESS, ''.join(stringList)`
The error case would be `return Errors.SOMEERROR, ''` (empty string)

File: project/eScavenge/Game.py [hunk 4]
```diff
@@ -597,6 +607,55 @@ def test_add_landmark2(self):
...
+        clue_time = datetime.timedelta(hours=0, minutes=20, seconds=50)
+        clue_time2 = datetime.timedelta(hours=0, minutes=10, seconds=20)
+        clue_time3 = datetime.timedelta(hours=0, minutes=2, seconds=0)
+        self.game._Game__teams["Team1"].time_log.append(now - clue_time)
+        self.game._Game__teams["Team1"].time_log.append(now - clue_time2)
+        self.game._Game__teams["Team1"].time_log.append(now - clue_time3)
+        self.game._Game__teams["Team1"].current_landmark = 2
+        self.game._Game__teams["Team1"].points = 10
+        self.assertEqual("Team: Team1\nYou Are On Landmark 3\nTime Taken For Landmarks: 19:12:35\nTotal Points: 10\n",
+                         self.game.get_snapshot(now), "Format incorrect")
```
Comment (holyzala) [inline project/eScavenge/Game.py]: To do the checking here (if you apply my comments) you would do something like
```Python
err, rtn = self.game.get_snapshot(now)
self.assertEqual(success, err)
self.assertEqual(string, run)
```

**Pull request merged**